### PR TITLE
Iteration 3 - invoice_item and invoice_item_repository 

### PR DESCRIPTION
**Pull request by KelsiePorter**
This PR includes invoice_item, invoice_item_repository, and associated passing tests.

### CODE REVIEW

File: lib/invoice_item.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    @item_id = invoice_item_data[:item_id]
+    @invoice_id = invoice_item_data[:invoice_id]
+    @quantity = invoice_item_data[:quantity]
+    @unit_price = BigDecimal((invoice_item_data[:unit_price].to_f / 100), 4)
+    @created_at = invoice_item_data[:created_at]
+    @updated_at = invoice_item_data[:updated_at]
+  end
+
+  def unit_price_to_dollars 
+    "$#{@unit_price.to_f.to_s}"
```
[the remark this answers is not in the document]
Comment (KelsiePorter) [reply lib/invoice_item.rb:25]: FYI - This method did not work without the `.to_s` included.

File: lib/invoice_item_repository.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+require_relative 'make_time'
+require_relative 'repository'
+
+class InvoiceItemRepository < Repository
```
[the remark this answers is not in the document]
Comment (Elle-M) [reply lib/invoice_item_repository.rb:3]: Is make_required here?

File: spec/invoice_item_repository_spec.rb [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                  :updated_at => Time.now
+                })
+      iir.create({ 
+                  :item_id => 24,
+                  :invoice_id => 25,
+                  :quantity => 1,
+                  :unit_price => BigDecimal("2099",4),
+                  :created_at => Time.now,
+                  :updated_at => Time.now
+                })
```
[the remark this answers is not in the document]
Comment (Elle-M) [reply spec/invoice_item_repository_spec.rb:109]: Did we decide on a consensus on hash and string syntex?

File: lib/invoice_item.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+              :created_at,
+              :updated_at
+
+  def initialize(invoice_item_data)
+    @id = invoice_item_data[:id]
+    @item_id = invoice_item_data[:item_id]
+    @invoice_id = invoice_item_data[:invoice_id]
+    @quantity = invoice_item_data[:quantity]
+    @unit_price = BigDecimal((invoice_item_data[:unit_price].to_f / 100), 4)
+    @created_at = invoice_item_data[:created_at]
```
Comment (this-is-joeking) [inline lib/invoice_item.rb]: Should we add #return_time_from()  to assignment on lines 20 and 21?

File: spec/invoice_item_repository_spec.rb [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+      iir.create({ 
+                  :item_id => 21,
+                  :invoice_id => 22,
+                  :quantity => 1,
+                  :unit_price => BigDecimal("1999",4),
+                  :created_at => Time.now,
+                  :updated_at => Time.now
+                })
+      iir.create({ 
+                  :item_id => 24,
```
[the remark this answers is not in the document]
Comment (Elle-M) [reply spec/invoice_item_repository_spec.rb:103]: indentation differences between classes?

**Comment by Elle-M** [comment]
Looks good to me! Just a couple minor potential adjustments.

**Pull request merged**